### ISSUE TITLE
Design time availability component

### ISSUE THREAD

**Issue by ryan-conn** [aggie-coding-club/Automatic-Aggie-Scheduler#121]
We need to create a way for users to select the time slots in which they want to take classes (similar to this):
![image](https://user-images.githubusercontent.com/28655462/71757976-e95a4700-2e5f-11ea-844c-a9a3f5689575.png)
We don't yet have a usable design for time selection, so we'll have to make one (if you'd like to do this, let us know and assign yourself). The only consideration as of now is that it will be going over the current schedule component, so we'll need a way to show/hide time selection when the user wants to, and the design needs to keep this in mind

**Comment by firejake308** [comment]
I've been messing around with this in a branch called `frontend/availability` and I'll try to post screenshots of some potential designs soon

**Comment by firejake308** [comment]
![localhost_8000_schedule(iPad)](https://user-images.githubusercontent.com/10082177/72099694-bcd27f00-32e6-11ea-9835-aba2e19245a9.png)
So I've got three sketches for how to display availability on the calendar: a solid red box, a red outlined box, or a red striped box. My favorite so far is the stripes. I'm working on setting up a click-and-drag function to create these boxes.

**Comment by gannonprudhomme** [comment]
Just a thought, but I think I'd like it to be semi-transparent to show the boundaries of the times (like how the when2meet site is). 

Also, using the color red makes me think that the highlighted times are times that you _don't_ want to have classes at, not times that you're available. Maybe use green? Idk if I'm just thinking that b/c when2meet uses green, but I'm ofc open to any color, just whatever seems intuitive.

**Comment by firejake308** [comment]
Lmao, I'm actually intentionally using red to show time that the user is not available right now. The main reason for this is so that I don't have to make the availability disappear when we add courses to the schedule, because otherwise the courses would be overlapping with green availability. It shouldn't be too hard to make the background transparent though, so I'll try that next time I'm working on it

**Comment by ryan-conn** [comment]
Apologies as I meant to post this as soon as the meeting was over, but this is the feedback I wrote during our last meeting:

- Should be able to see what time you’re hovering over (show a horizontal line at the current time and display what time it is on the left side?)
- Should be able to mark multiple days at a time
- Shouldn’t be able to click the same starting/ending time (should probably default to 10 minute block of time)
- Show trash can icon (could just replace the busy text) when hovering over a block, and click on it to delete
- Right clicking currently makes a block, but it shouldn't
- Automatically merge overlapping blocks
- Make blocks semitransparent (`#f4433680`?)

**Comment by firejake308** [comment]
I elected to set the minimum time to 30 minutes, as 10 minutes renders as a single, ugly line rather than a beautiful box. I also fixed right-clicking.

**Comment by firejake308** [comment]
Also making the blocks semitransparent is super easy, but I'm going to wait until the schedule background PR is merged so that we can get a better idea of what it actually looks like

**Comment by gannonprudhomme** [comment]
![chrome_ckG8WJEOab](https://user-images.githubusercontent.com/7295783/72497627-98006f00-37f2-11ea-81ee-8d3f3082ce74.png)

Looks fine for me

**Comment by firejake308** [comment]
Added merging time blocks and updated the background color of the availability cards. I think the new color looks good.

**Comment by firejake308** [comment]
If you look at the latest version of the `frontend/availability` branch, you'll see that I've implemented live update. I've also preserved the merging behavior from previous commits and kept the minimum size of a block at 30 minutes, so left clicking once will create a 30-minute block. 

Do y'all think it's fine if the blocks automatically merge during a live update, or would it be better if the blocks overlapped as long as you held the mouse down, and then they merged once you let go?
